tests/derive/test_assign_datetime_from_group.py

```diff
@@ -0,0 +1,38 @@
+from chispa import assert_df_equality
+from pyspark.sql import functions as F
+
+from cishouseholds.derive import assign_datetime_from_group
+
+
+def test_assign_datetime_from_group(spark_session):
+    expected_df = spark_session.createDataFrame(
+        data=[
+            ("2020-04-18 00:00:00", "2020-04-18 11:59:59", "2020-07-20 05:30:00", "2020-04-18 12:00:00"),
```

Comment:
If the first column here was a datetime, not a date, we would be removing a genuine time. We need to only adjust the time if a date field is chosen